app/forms/idv/api_image_upload_form.rb

```diff
@@ -84,7 +84,20 @@ def validate_image(image_key)
       file.rewind
 
       return if file.content_type.start_with?('image/') && data.present?
-      errors.add(image_key, t('doc_auth.errors.must_be_image'))
+      error = error_from_yaml(data) if !Rails.env.production? && yaml_file?(file)
+      error ||= t('doc_auth.errors.must_be_image')
```

Comment:
1. I think we want this kind of thing to work in INT as well, right? To be able to trigger errors in int? Int counts as "production" so I think the check we want is more like: `LoginGov::Hostdata.domain != 'login.gov'` because it seprates staging & production (`login.gov` domain)` from everything else

2. If we wanted to make the guard clearer and combine I think we could di:

```suggestion
      if LoginGov::Hostdata.domain != 'login.gov' && yaml_file?(file)
        error = error_from_yaml(data) ||  t('doc_auth.errors.must_be_image')
        errors.add(image_key, error)
      end
```

but rubocop will get mad at the guard class 🙄  so it'll probably end up more like

```suggestion
      return if LoginGov::Hostdata.domain == 'login.gov' || !yaml_file?(file)
      error = error_from_yaml(data) ||  t('doc_auth.errors.must_be_image')
      errors.add(image_key, error)
```

Reply:
> 1. I think we want this kind of thing to work in INT as well, right? To be able to trigger errors in int? Int counts as "production" so I think the check we want is more like: `LoginGov::Hostdata.domain != 'login.gov'` because it seprates staging & production (`login.gov` domain)` from everything else

Tying this back to a comment from my original post and how it's implemented today, I wonder if this is something where the switch should occur based on the implementation of the DocAuth client (`'mock'` vs. `'acuant'`). The previous flow would support this merely by the implementation of errors handling in the mock client ([`DocAuth::Mock::ResultResponseBuilder`](https://github.com/18F/identity-idp/blob/master/app/services/doc_auth/mock/result_response_builder.rb)).

So this could be something as simple as:

```suggestion
      error = error_from_yaml(data) if DocAuth::Client.doc_auth_vendor == 'mock' && yaml_file?(file)
      error ||= t('doc_auth.errors.must_be_image')
```

That being said, I'm beginning to wonder if this is something where we shouldn't need to be implementing this separately, but instead taking advantage of existing errors handling from the mock client.

The problem is that we're considering this validation before (and as a blocker for) [the client submission where those errors would be read from the YAML file](https://github.com/18F/identity-idp/blob/697dd56df9d72628469600a9ad8784503bc629ab/app/controllers/idv/image_uploads_controller.rb#L12-L18).

Also, unless I'm mistaken, [we're only considering YAML errors from the `back` image](https://github.com/18F/identity-idp/blob/697dd56df9d72628469600a9ad8784503bc629ab/app/services/doc_auth/mock/doc_auth_mock_client.rb#L81), which may be fine/expected, but wasn't clear to me, nor is mentioned in the testing documentation.

What would be the downsides to removing `ApiImageUploadForm` validation altogether? Or the entire class? Or at least moving some of those simple checks ("is image") into the client implementation? Is it any different from [how we're handling these values in the pre-React flow](https://github.com/18F/identity-idp/blob/697dd56df9d72628469600a9ad8784503bc629ab/app/services/idv/steps/doc_auth_base_step.rb#L100-L112)?

Reply:
I think you're right, there is a good case for removing the file checks and delegating entirely to the client. We should make sure the client is robust enough to handle bad input, but otherwise yes, it would simplify a lot.

Reply:
A bit of thinking out loud: Separately, I've been looking at rate-limiting (LG-3342) which fits a similar need of "cross-client" validations, which could justify the continued existence of `ApiImageUploadForm`. Both validations ("is image", "is throttled") would need to be accounted for in each client implementation (base class? mixins?). Another point is whether clients should be aware of a particular user, or instead only responsible for managing communicating to the doc auth vendor. It doesn't apply for the "is image" validations, but we'd need to associate a user for "is throttled".

Reply:
I agree, I wouldn't want the client classes to have to worry about a particular user

Throttling does seem like a good reason to keep the form, maybe we just make it a throttling form or something? We can always expand if there are new validations in the future. Or just remove the form in this PR and add it back when rate limiting comes back?

Reply:
Yeah, I'm in favor of dropping the validations from the form and handle it, handle it on the frontend, and letting the clients fail if the user does manage to work something unexpected in.

If you know you're gonna use the forms for LG-3342, we could work in an empty `submit` function here so we don't have to pull it out and put it right back in.

Reply:
There's also a few other validations in this form we want, notably the `validates_presence_of`.

In b6389f8, I removed the validation to check that the file is an image file, to allow this to pass through to the client implementation to handle.

I'd considered removing `validate_images` altogether, but since we [proceed to call `.read` on them](https://github.com/18F/identity-idp/blob/752316f75920e5c6c61f14c9cf11e02ebe9b6c11/app/controllers/idv/image_uploads_controller.rb#L14), we'd want to at least ensure that they're files. I've updated the check in the commit to consider `respond_to?(:read)` (previously `:content_type`) to match.